lib/portage/util/__init__.py

```diff
@@ -1505,15 +1509,26 @@ def close(self):
         """Closes the temporary file, copies permissions (if possible),
         and performs the atomic replacement via os.rename().  If the abort()
         method has been called, then the temp file is closed and removed."""
-        f = object.__getattribute__(self, "_file")
-        tmp_name = object.__getattribute__(self, "_tmp_name")
-        real_name = object.__getattribute__(self, "_real_name")
-        if not f.closed:
+        try:
+            f = object.__getattribute__(self, "_file")
+        except AttributeError:
+            f = None
+        if f is not None and not f.closed:
+            real_name = object.__getattribute__(self, "_real_name")
+            tmp_name = object.__getattribute__(self, "_tmp_name")
             try:
                 f.close()
                 if not object.__getattribute__(self, "_aborted"):
                     try:
-                        apply_stat_permissions(tmp_name, os.stat(real_name))
+                        try:
+                            st = os.stat(real_name)
+                        except FileNotFoundError:
+                            umask_test_file = f"{tmp_name}_umask_test"
```

Comment:
I've realized this is unsafe for bytes tmp_name, much like `tmp_name = "%s.%i" % (canonical_path, portage.getpid())` mentioned in https://github.com/gentoo/portage/pull/1132#discussion_r1360120482.